### PR TITLE
[MRG] Release 0.9.4

### PR DESCRIPTION
**Pull request by rflamary**
## PR checklist

- [x] I have read the [**CONTRIBUTING**](CONTRIBUTING.md) document.
- [x] The documentation is up-to-date with the changes I made (check build artifacts).
- [x] All tests passed, and additional code has been **covered with new tests**.
- [x] I have added the PR and Issue fix to the [**RELEASES.md**](RELEASES.md) file.

### CODE REVIEW

File: RELEASES.md [hunk 1]
```diff
@@ -1,6 +1,12 @@
...
+
+From a maintenance point of view, we now have a new option to install optional dependencies with `pip install POT[all]` and the specific backends or submodules' dependencies may also be installed individually. The pip options are: `backend-jax, backend-tf, backend-torch, cvxopt, dr, gnn, all`. We lso provide with this release support for NumPy 2.0 (the wheels should now be compatible with NumPy 2.0 and below). We also fixed several issues such as gradient sign errors for FGW solvers, empty weights for `ot.emd2`, and line-search in partial GW. We also split the `test/test_gromov.py` into `test/gromov/` to make the tests more manageable.
```
Comment (matthewfeickert) [inline RELEASES.md:9]:
```suggestion
From a maintenance point of view, we now have a new option to install optional dependencies with `pip install POT[all]` and the specific backends or submodules' dependencies may also be installed individually. The pip options are: `backend-jax, backend-tf, backend-torch, cvxopt, dr, gnn, all`. We also provide with this release support for NumPy 2.0 (the wheels should now be compatible with NumPy 2.0 and below). We also fixed several issues such as gradient sign errors for FGW solvers, empty weights for `ot.emd2`, and line-search in partial GW. We also split the `test/test_gromov.py` into `test/gromov/` to make the tests more manageable.
```

Comment (matthewfeickert) [reply RELEASES.md:9]: Not trying to be annoying here and edit your release notes, but I noticed a tiny tiny typo.

Comment (rflamary) [reply RELEASES.md]: this is still definitely  WIP but I appreciate the correction.

**Comment by rflamary** [comment]
OK this should be the one. Will merge it when all pass and wait for the wheel build  before doing the proper release

**Comment by rflamary** [comment]
@matthewfeickert thanks for your help!

I have released 0.9.4 on pipy and github. conda is coming but will take a few more hours/days.

**Comment by matthewfeickert** [comment]
Fantastic! Thank you so much @rflamary and @cedricvincentcuaz! I very much appreciate how helpful you've been and how quickly you moved to get a release out. :rocket:

**Pull request merged**